App.js

```diff
@@ -153,28 +181,7 @@ class MonthView extends React.Component {
                     hideArrows = {true}
                     markingType = "multi-dot"
                     markedDates = {this._generateMarkedDates()}
-                    onVisibleMonthsChange = {date => {
-                        this.setState({
-                            selectedMonth: date[0] ? date[0].month - 1 : this.state.selectedMonth,
-                            selectedYear: date[0] ? date[0].year : this.state.selectedYear
-                        });
-
-                        let backupDate = this.state.selectedYear + "-" + this.state.selectedMonth + 1;
-
-                        let fullDate = date[0] ? date[0].dateString : backupDate;
-                        let monthString = fullDate.substring(0, 7);
-
-                        if (!loadedData.viewedMonths.includes(monthString) && monthString < loadedData.viewedMonths[0]) {
-                            loadedData.viewedMonths.push(monthString);
-
-                            db.getMonthEvents(fullDate).then(events => {
-                                if (events.length > 0) {
-                                    loadedData.events = _.unionBy(loadedData.events, events, "id");
-                                    this.forceUpdate();
-                                }
-                            }).catch(err => console.log(err));
-                        }
-                    }}
+                    onVisibleMonthsChange = {this._visibleMonthsChange}
```

Comment:
Something to remember with JavaScript is that its scoping for "this" is oftentimes very different from what you'd expect. Unless I'm mistaken, this line won't execute correctly. You either need to bind the function to the class within its constructor, or convert this reference to an arrow function that simply calls the class function.

Reply:
I tested it on Expo, and experienced no issues or abnormal behavior... but you're right.

It's better to avoid problems with "this," both for us and any team that picks up this project later.  I'll push the change up in a moment.